### DISCUSSION TITLE
Attachment removed from another model if same attachment attach with same model id;

### DISCUSSION THREAD

**Discussion by vinod-digitalcubez** [orchidsoftware/platform#2706] [General]
I have two models with the same ID and I sync the attachment, but when I remove the attachment from one model, it also deletes from another model attachment.

**Comment by macit-emre** [comment]
its about you have to learn manyToMany relations in laravel. 
and if you want to workaround, make a copy of your attachment instead. 
this is made by : 
go to config/platform.php and change 
``` 
'attachment' => [
        'disk'      => env('FILESYSTEM_DISK', 'public'),
        //'generator' => \Orchid\Attachment\Engines\Generator::class, //change this to your own generator like me.
        'generator' => \App\Orchid\Generators\Generator::class,
    ],
```

and inside your own generator use 
```
 public function hash(): string
    {
        return parent::hash().rand(100000,999999); //or some other random number generation mechanism.
    }
```
